extensions/replication/replicationStatusProcessor/ReplicationStatusProcessor.js

```diff
@@ -150,7 +150,11 @@ class ReplicationStatusProcessor {
         if (backend) {
             const bucket = queueEntry.getBucket();
             const key = queueEntry.getObjectKey();
-            const versionId = queueEntry.getEncodedVersionId();
+            // If the original CRR was on a bucket without versioning enabled
+            // (i.e. an NFS bucket), maintain the Redis key schema by using
+            // and empty string.
+            const versionId = queueEntry.getVersionId() ?
```

Comment:
Since you're now doing the `getVersionId()` check every time to get the encoded version, why not implementing this directly into `getEncodedVersionId()`.